tests/core/math/test_math_funcs.h

```diff
@@ -0,0 +1,556 @@
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef TEST_MATH_FUNCS_H
+#define TEST_MATH_FUNCS_H
+
+#include "tests/test_macros.h"
+
+namespace TestMath {
+
+TEST_CASE("[Math] C++ macros") {
+	CHECK(MIN(-2, 2) == -2);
+	CHECK(MIN(600, 2) == 2);
+
+	CHECK(MAX(-2, 2) == 2);
+	CHECK(MAX(600, 2) == 600);
+
+	CHECK(CLAMP(600, -2, 2) == 2);
+	CHECK(CLAMP(620, 600, 650) == 620);
+	// `max` is lower than `min`.
+	CHECK(CLAMP(620, 600, 50) == 50);
+
+	CHECK(ABS(-5) == 5);
+	CHECK(ABS(0) == 0);
+	CHECK(ABS(5) == 5);
+
+	CHECK(SIGN(-5) == doctest::Approx(-1.0));
+	CHECK(SIGN(0) == doctest::Approx(0.0));
+	CHECK(SIGN(5) == doctest::Approx(1.0));
+}
+
+TEST_CASE("[Math] Power of two functions") {
+	CHECK(next_power_of_2(0) == 0);
+	CHECK(next_power_of_2(1) == 1);
+	CHECK(next_power_of_2(16) == 16);
+	CHECK(next_power_of_2(17) == 32);
+	CHECK(next_power_of_2(65535) == 65536);
+
+	CHECK(previous_power_of_2(0) == 0);
+	CHECK(previous_power_of_2(1) == 1);
+	CHECK(previous_power_of_2(16) == 16);
+	CHECK(previous_power_of_2(17) == 16);
+	CHECK(previous_power_of_2(65535) == 32768);
+
+	CHECK(closest_power_of_2(0) == 0);
+	CHECK(closest_power_of_2(1) == 1);
+	CHECK(closest_power_of_2(16) == 16);
+	CHECK(closest_power_of_2(17) == 16);
+	CHECK(closest_power_of_2(65535) == 65536);
+
+	CHECK(get_shift_from_power_of_2(0) == -1);
+	CHECK(get_shift_from_power_of_2(1) == 0);
+	CHECK(get_shift_from_power_of_2(16) == 4);
+	CHECK(get_shift_from_power_of_2(17) == -1);
+	CHECK(get_shift_from_power_of_2(65535) == -1);
+
+	CHECK(nearest_shift(0) == 0);
+	CHECK(nearest_shift(1) == 1);
+	CHECK(nearest_shift(16) == 5);
+	CHECK(nearest_shift(17) == 5);
+	CHECK(nearest_shift(65535) == 16);
+}
+
+TEST_CASE("[Math] abs") {
+	// int
+	CHECK(Math::abs(-1) == 1);
+	CHECK(Math::abs(0) == 0);
+	CHECK(Math::abs(1) == 1);
+
+	// double
+	CHECK(Math::abs(-0.1) == doctest::Approx(0.1));
+	CHECK(Math::abs(0.0) == doctest::Approx(0.0));
+	CHECK(Math::abs(0.1) == doctest::Approx(0.1));
+
+	// float
+	CHECK(Math::abs(-0.1f) == doctest::Approx(0.1f));
+	CHECK(Math::abs(0.0f) == doctest::Approx(0.0f));
+	CHECK(Math::abs(0.1f) == doctest::Approx(0.1f));
+}
+
+TEST_CASE("[Math] round/floor/ceil") {
+	CHECK(Math::round(1.5) == doctest::Approx(2.0));
+	CHECK(Math::round(1.6) == doctest::Approx(2.0));
+	CHECK(Math::round(-1.5) == doctest::Approx(-2.0));
+	CHECK(Math::round(-1.1) == doctest::Approx(-1.0));
+
+	CHECK(Math::floor(1.5) == doctest::Approx(1.0));
+	CHECK(Math::floor(-1.5) == doctest::Approx(-2.0));
+
+	CHECK(Math::ceil(1.5) == doctest::Approx(2.0));
+	CHECK(Math::ceil(-1.9) == doctest::Approx(-1.0));
```

Comment:
These should be tested to have exact results, since they are trivial operations with no rounding errors.

Reply:
Done :)